### ISSUE TITLE
Search Results page has improper heading hierarchy

### ISSUE THREAD

**Issue by alegontarz** [DSpace/dspace-angular#5500]
## Describe the bug
The Search Results page uses headings in an incorrect order. In the current structure, level 3 and level 4 headings appear before the main page heading, and the hierarchy is not logically nested.

From the screenshot, examples include:

H3: Filters
H3: Advanced Search
H4: Filter by
H3: Settings
H4: Sort By
H4: Results per page
H1: Search Results

This makes the heading structure confusing for screen reader users and keyboard users who rely on headings to understand the page layout and navigate efficiently.

<img width="2048" height="782" alt="Image" src="https://github.com/user-attachments/assets/8cea5a6f-ee08-4e9c-80d9-b3ee3b9d639e" />

## To Reproduce
1. Go to the search page
2. Use e.g. headings extension.

## Expected behavior
The page should have a clear, hierarchical heading structure, for example:

H1: Search Results
H2: Filters
H3: Advanced Search
H3: Filter by
H2: Settings
H3: Sort By
H3: Results per page

The exact structure may vary, but heading levels should reflect the visual and semantic organization of the page.

## Related
Past ticket to do similar header hierarchy resolution: https://github.com/DSpace/dspace-angular/issues/3071

**Comment by lgeggleston** [comment]
Thank you for raising! This would be a good issue to address for accessibility. 
Marking as needing discussion before someone picks it up though, just because I think there will need to be more clarity and agreement about what a good header structure choice would be before it's implemented.  In addition, since it's not just the type of headers but the order, I'm wondering if this would require some HTML/Angular component restructuring and how to minimize that if so.

**Comment by lgeggleston** [comment]
This ticket was discussed at today's [Developer meeting,](https://wiki.lyrasis.org/display/DSPACE/2026-04-23+DSpace+Developers+Meeting) and most notably anyone looking into this could start from previous similar work on header hierarchy found here: https://github.com/DSpace/dspace-angular/issues/3071 

Additional comments:
- There are differing opinions on what hierarchy should look like; If there is anyone who uses a screen reader regularly or has more expertise, would be good to get input on proposed structure
- Filter could logically be separate from Search Results; however only there can only be one H1

**Comment by mwoodiupui** [comment]
I've thought a bit more about my off-the-cuff remark in the meeting.  I would suggest:

* The page is about Search, so the level 1 heading should be Search.
* Filters, Settings and Results are distinct aspects of a search.  Each should be level 2.  Results are what you get; Filters are how you choose what you get; Settings are about how to present what you get.

I note that the page is not solely about search *results* since it contains controls for requesting a new search.
